src/app/features/activity-list/components/submitted-transactions-list.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react';
+import { Box, color, Stack, Text } from '@stacks/ui';
+
+import { SubmittedTransactionsItem } from '@app/features/activity-list/components/submitted-transactions-item';
+import {
+  useCleanupSubmittedTxsCallback,
+  useSubmittedStacksTransaction,
+} from '@app/store/accounts/submitted-transactions.hooks';
+import { SubmittedTransaction } from '@app/store/accounts/submitted-transactions';
+
+const SubmittedTransactionsListItem = ({ txid }: { txid: string }) => {
+  const submittedTx = useSubmittedStacksTransaction(txid);
+  if (submittedTx) return <SubmittedTransactionsItem transaction={submittedTx} txid={txid} />;
+  // TODO: Use a better loading state?
+  return <>Loading...</>;
+};
+
+interface SubmittedTransactionsListProps {
+  txs: SubmittedTransaction[];
+}
+export const SubmittedTransactionsList = ({ txs }: SubmittedTransactionsListProps) => {
+  const cleanup = useCleanupSubmittedTxsCallback();
+
+  useEffect(() => cleanup(), [cleanup]);
```

Comment:
This part of the implementation I find unusual in that, what does the component life cycle have to do with the submitted transaction state?

It'd be better if the two were not coupled. Changing/moving around components, a different design/ux of the tx list, could alter when this is called.

Reply:
Yeah, agree, I wanted to get stuff working but wasn't clear on why this is here so I'll rethink it.